semver/_shared.ts

```diff
@@ -63,78 +63,89 @@ export function compareIdentifier(
   return 0;
 }
 
-// The following Regular Expressions can be used for tokenizing,
-// validating, and parsing SemVer version strings.
-
-// ## Numeric Identifier
-// A single `0`, or a non-zero digit followed by zero or more digits.
-
+/**
+ * A single `0`, or a non-zero digit followed by zero or more digits.
+ */
 const NUMERIC_IDENTIFIER = "0|[1-9]\\d*";
 
-// ## Non-numeric Identifier
-// Zero or more digits, followed by a letter or hyphen, and then zero or
-// more letters, digits, or hyphens.
-
+/**
+ * Zero or more digits, followed by a letter or hyphen, and then zero or more letters, digits, or hyphens.
+ */
 const NON_NUMERIC_IDENTIFIER = "\\d*[a-zA-Z-][a-zA-Z0-9-]*";
 
-// ## Main Version
-// Three dot-separated numeric identifiers.
-const MAIN_VERSION =
+/**
+ * Three dot-separated numeric identifiers.
+ */
+const VERSION_CORE =
   `(?<major>${NUMERIC_IDENTIFIER})\\.(?<minor>${NUMERIC_IDENTIFIER})\\.(?<patch>${NUMERIC_IDENTIFIER})`;
 
-// ## Pre-release Version Identifier
-// A numeric identifier, or a non-numeric identifier.
-
+/**
+ * A numeric identifier, or a non-numeric identifier.
+ */
 const PRERELEASE_IDENTIFIER =
   `(?:${NUMERIC_IDENTIFIER}|${NON_NUMERIC_IDENTIFIER})`;
 
-// ## Pre-release Version
-// Hyphen, followed by one or more dot-separated pre-release version
-// identifiers.
-
+/**
+ * A Hyphen, followed by one or more dot-separated pre-release version identifiers.
+ * @example "-pre.release"
+ */
 const PRERELEASE =
   `(?:-(?<prerelease>${PRERELEASE_IDENTIFIER}(?:\\.${PRERELEASE_IDENTIFIER})*))`;
 
-// ## Build Metadata Identifier
-// Any combination of digits, letters, or hyphens.
+/**
+ * Any combination of digits, letters, or hyphens.
+ */
 const BUILD_IDENTIFIER = "[0-9A-Za-z-]+";
 
-// ## Build Metadata
-// Plus sign, followed by one or more period-separated build metadata
-// identifiers.
+/**
+ * A plus sign, followed by one or more period-separated build metadata identifiers.
+ * @example "+build.meta"
+ */
 const BUILD =
   `(?:\\+(?<buildmetadata>${BUILD_IDENTIFIER}(?:\\.${BUILD_IDENTIFIER})*))`;
 
-// ## Full Version String
-// A main version, followed optionally by a pre-release version and
-// build metadata.
-
-// Note that the only major, minor, patch, and pre-release sections of
-// the version string are capturing groups.  The build metadata is not a
-// capturing group, because it should not ever be used in version
-// comparison.
-const FULL_PLAIN = `v?${MAIN_VERSION}${PRERELEASE}?${BUILD}?`;
+/**
+ * A version, followed optionally by a pre-release version and build metadata.
+ */
+const SEMVER = `v?${VERSION_CORE}${PRERELEASE}?${BUILD}?`;
```

Comment:
I think how this line was before was fine. All the other changes look good.

Reply:
Do you mean the comment about capturing groups? Because `BUILD` is using a capturing group and is used for parsing. It is not used for comparisons, but that logic is inside `compare()`.

Reply:
Sorry, I just meant how the constant and its parts are named.

Reply:
Ah, I see. I don't know, I think `FULL_PLAIN` is not very descriptive. The comment even had to clarify: `## Full Version String` and `A main version, ...`. Maybe `SEMVER_VERSION` or `FULL_VERSION`?

Reply:
`FULL_VERSION` SGTM

Reply:
Done